### PR TITLE
[Feature] Support dql 'show catalogs' for dla connector, support add comment when executing catalog ddl

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/5962

## Problem Summary(Required) ：

1. Support SQL like 'Show catalogs' for dla connector.
2. Support add comment for ddl create external catalog.Comment is optional, and no comment won't lead to failure.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 59 (47.46%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|15|06.67%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCatalogsStmt.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/persist/CreateCatalogLog.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 1]
```diff
@@ -27,6 +27,7 @@ BY: 'BY';
 CASE: 'CASE';
 CAST: 'CAST';
 CATALOG: 'CATALOG';
+CATALOGS: 'CATALOGS';
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:30]: make it non reserved

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4]: > make it non reserved

done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 59 (47.46%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|15|06.67%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ShowCatalogsStmt.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/persist/CreateCatalogLog.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/persist/CreateCatalogLog.java [hunk 2]
```diff
@@ -16,18 +16,25 @@ public class CreateCatalogLog implements Writable {
...
     @SerializedName(value = "catalogName")
     private String catalogName;
+    @SerializedName(value = "comment")
+    private String comment;
     @SerializedName(value = "properties")
     private Map<String, String> properties;
 
-    public CreateCatalogLog(String catalogName, Map<String, String> properties) {
+    public CreateCatalogLog(String catalogName, String comment, Map<String, String> properties) {
         this.catalogName = catalogName;
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/persist/CreateCatalogLog.java:25]: What if I don't want to add any comments? will this catalog create with failure?

**Review by caneGuy — COMMENTED** [review]
we should describe that comment field was added in the details

File: fe/fe-core/src/main/java/com/starrocks/persist/CreateCatalogLog.java (hunk 2 above)
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/persist/CreateCatalogLog.java:25]: > What if I don't want to add any comments? will this catalog create with failure?

MySQL [demodb]> CREATE EXTERNAL CATALOG hive_catalog_1 COMMENT "hive_catalog" PROPERTIES("type"="hive", "hive.metastore.uris"="thrift://127.0.0.1:9083");
Query OK, 0 rows affected (0.00 sec)

MySQL [demodb]> CREATE EXTERNAL CATALOG hive_catalog_2 PROPERTIES("type"="hive", "hive.metastore.uris"="thrift://127.0.0.1:9084");
Query OK, 0 rows affected (0.00 sec)

MySQL [demodb]> show catalogs;
+----------------+------+--------------+
| Catalog        | Type | Comment      |
+----------------+------+--------------+
| hive_catalog_2 | hive | NULL         |
| hive_catalog_1 | hive | hive_catalog |
+----------------+------+--------------+
2 rows in set (0.00 sec)

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/persist/CreateCatalogLog.java:25]: > What if I don't want to add any comments? will this catalog create with failure?

If no comment is announced when executing create catalog ddl, comment would be NULL.It won't lead to failure.

**Review by caneGuy — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 3]
```diff
@@ -38,6 +38,8 @@ statement
     // Catalog Statement
     | createExternalCatalogStatement                                                        #createCatalog
     | dropExternalCatalogStatement                                                          #dropCatalog
+    | showCatalogStatement                                                                  #showCatalogs
+    | showDbFromCatalogStatement                                                            #showDbFromCatalog
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:42]: you need adapt it on line "52" #showDatabases

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 4]
```diff
@@ -145,13 +147,21 @@ alterSystemStatement
...
 
 dropExternalCatalogStatement
     : DROP EXTERNAL CATALOG catalogName=identifierOrString
     ;
 
+showCatalogStatement
+    : SHOW CATALOGS
+    ;
+
+showDbFromCatalogStatement
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:161]: ditto

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 5]
```diff
@@ -1547,4 +1553,20 @@ private void handleShowWorkGroup() {
...
+
+    private void handleShowCatalogs() {
+        ShowCatalogsStmt showCatalogsStmt = (ShowCatalogsStmt) stmt;
+        ConcurrentHashMap<String, Catalog> catalogs = GlobalStateMgr.getCurrentState().getCatalogMgr().getCatalogs();
+        List<List<String>> rows = Lists.newArrayList();
+        for (Map.Entry<String, Catalog> entry : catalogs.entrySet()) {
+            List<String> catalog = Lists.newArrayList();
+            catalog.add(entry.getKey());
+            Catalog value = entry.getValue();
+            catalog.add(((ExternalCatalog) value).getType());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:1565]: add "default" as internal catalog

**Pull request merged**